core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -518,6 +518,17 @@ class Partition(val topic: String,
     info
   }
 
+  def logStartOffset : Long = {
+    inReadLock(leaderIsrUpdateLock) {
+      leaderReplicaIfLocal match {
+        case Some(leaderReplica) =>
+          return leaderReplica.log.get.logStartOffset
+        case None =>
+          -1
+      }
```

Comment:
`leaderReplicaIfLocal.map(_.log.get.logStartOffset).getOrElse(-1)` is more concise.